Assets/Scripts/LocksAndKeys/Keys/DialKey.cs

```
@@ -0,0 +1,381 @@
+                }
+                notches.RemoveRange(NotchCount, notches.Count - NotchCount);
+            }
+            else if (notches.Count < NotchCount)
+            {
+                int notchListCount = notches.Count;
+                for (int i = 0; i < NotchCount - notchListCount; i++)
+                {
+                    DialNotch newNotch = Instantiate(notchPrefab, notchParent).GetComponent<DialNotch>();
+                    int notchNum = startNum + i + notchListCount;
+					newNotch.Setup(DEMO_RADIUS, 20, usingChars ? ((char)notchNum).ToString() : notchNum.ToString());
+					notches.Add(newNotch);
+                }
+            }
+
+            for (int i = 0; i < notches.Count; i++)
+            {
+                DialNotch notch = notches[i];
+                notch.SetPositionAndRotation(i, notches.Count);
+            }
+			updateCurNotch();
+		}
+
+		private bool notchUpdateRequired()
+        {
+            return notches.Count != NotchCount || charsIsDirty;
+		}
+
+        private int convertLowerCharToUpper(int character)
+        {
+            return character + 'Z' - 'A' + 1;
+
+		}
+        
+        // WARNING! Lots of magic numbers below here.
+
+        private void rotateByMouse()
+        {
+            int modX = Input.mousePosition.y >= transform.position.y ? -1 : 1;
+            rotateMechanism(modX * Input.mousePositionDelta.x / Screen.width * 360);
+        }
+
+        private void rotateByScroll()
+        {
+            rotateMechanism(Input.mouseScrollDelta.y * 3);
+        }
+
+        private void rotateByKey(bool clockwise)
+        {
+            int mod = clockwise ? -1 : 1;
+            rotateMechanism(0.25f * mod);
+        }
+
+        private void rotateMechanism(float degrees)
+        {
+            transform.Rotate(0, 0, degrees);
+            if (degrees != 0)
+            {
+				if (curDir == 0)
+				{
+                    int dir = (int)(-degrees / Mathf.Abs(degrees));
```

should be an issue, but be careful of division by zero.

---

I'd like to avoid a try catch here. If we are dividing by zero, I want as many errors to fire as possible because we should not be doing that anywhere ever. A try catch would only serve to suppress the error.

I'm adding a comment to the if check for a degrees value of zero so that any deranged programmer (i.e. one of us) tries to put the logic outside of the if check, they know what they're in for.